src/app/content/utils.ts

```diff
@@ -30,11 +31,17 @@ export interface ContentPageRefencesType {
 
 const hashRegex = `#[^'"]+`;
 const pathRegex = `\\./((?<bookId>[a-z0-9-]+)(@(?<bookVersion>[^/]+))?):(?<pageId>[a-z0-9-]+)\\.xhtml(${hashRegex})?`;
-const referenceRegex = `['"]{1}(?<matchPath>((${pathRegex})|(${hashRegex})))`;
+const referenceRegex = `(?<matchPath>((${pathRegex})|(${hashRegex})))`;
 
 export function getContentPageReferences(book: ArchiveBook, page: ArchivePage) {
+  const domParser = new DOMParser();
+  const domNode = domParser.parseFromString(page.content, 'text/html');
+  const references = Array.from(domNode.links)
+    .map((link) => (link as HTMLAnchorElement).getAttribute('href') || '')
+    .filter((href) => href && href.match(new RegExp(referenceRegex, 'g')));
+
   const matches: ContentPageRefencesType[] = (
-    page.content.match(new RegExp(referenceRegex, 'g')) || []
+    references
   )
     .map((match) => match.match(new RegExp(referenceRegex))?.groups)
```

Comment:
you can combine these method chains, you're just staring with the result of the previous one here